syft/frameworks/crypten/jail.py

```diff
@@ -0,0 +1,133 @@
+import torch
+import crypten
+import syft
+import re
+import ast
+from dill.source import getsource
+from RestrictedPython import compile_restricted, safe_builtins
+from RestrictedPython.Eval import default_guarded_getiter, default_guarded_getitem
+from RestrictedPython.Guards import (
+    guarded_iter_unpack_sequence,
+    guarded_unpack_sequence,
+    guarded_setattr,
+)
+from RestrictedPython.PrintCollector import PrintCollector
+
+
+class JailRunner:
+    """Execution envrionment with limited capabilities."""
+
+    def __init__(
+        self,
+        func_src=None,
+        func=None,
+        modules=[torch, syft, crypten],
+        rm_decorators=True,
+        **global_kwargs,
+    ):
+        """
+        Control what should be accesible from inside the function.
+
+        Args:
+            func_src: function's src to be jailed. Required if func isn't set.
+            func: function to be jailed. Ignored if func_src is set.
+            modules: python modules that should be accessible.
+            rm_decorators: specify if decorators should be removed. Default to True.
+            global_kwargs: globals to be accessible.
+        """
+
+        if func_src is None:
+            if func is None:
+                raise ValueError("Either func_src or func must be set")
+            else:
+                try:
+                    func_src = getsource(func)
+                except:  # use inspect if dill fail
+                    import inspect
+
+                    func_src = inspect.getsource(func)
+
+        if rm_decorators:
+            # Remove decorator if any
+            func_src = re.sub(r"@[^\(]+\([^\)]*\)", "", func_src)
+
+        is_func, self._func_name = JailRunner._check_func_def(func_src)
+        if not is_func:
+            raise ValueError("Not valid function definition")
+
+        self._func_src = func_src
+
+        self._jail_globals = global_kwargs
+        # save names for serialization
+        self._module_names = []
+        # add modules
+        for module in modules:
+            self._jail_globals[module.__name__] = module
+            self._module_names.append(module.__name__)
+
+        self._is_built = False
+        self._build()
+
+    def _build(self):
+        if self._is_built:
+            raise RuntimeWarning("JailRunner already built.")
+
+        exec_globals = self._jail_globals
+        exec_globals["__builtins__"] = safe_builtins
+        exec_globals["_getiter_"] = default_guarded_getiter
+        exec_globals["_getitem_"] = default_guarded_getitem
+        exec_globals["_getattr_"] = getattr
+        # for a, b in
```

Comment:
Remove comment?

Reply:
The comments are here explaining why a certain global attribute is set